package/MDAnalysis/core/groups.py

```diff
@@ -148,15 +148,21 @@ def make_classes():
 
     # The 'GBase' middle man is needed so that a single topologyattr
     #  patching applies automatically to all groups.
-    GBase = bases[GroupBase] = _TopologyAttrContainer._subclass()
+    GBase = bases[GroupBase] = _TopologyAttrContainer._subclass(singular=False)
+    GBase._SETATTR_WHITELIST = {  # list of Group attributes we can set
```

Comment:
When we create a new version of the GroupBase, we also create a fresh list of 'allowed' attributes